ci_framework/roles/devscripts/README.md

```diff
@@ -0,0 +1,235 @@
+| boot_mac_addr | MAC address of physical system connected to provisioning network. |
+| boot_mode | The mode to be used for booting. Choices are `legacy \| UEFI \| UEFISecureBoot`. |
+| extra_spec | Key/value pairs as supported by [baremetal-operator](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md) |
+
+#### Supported keys in bmc
+
+The keys supported in `cifmw_devscripts_osp_compute_nodes.bmc` are
+
+| Key | Description |
+| --- | ----------- |
+| address | URL to the servers BMC. Refer notes for additional information. |
+| username | Name of the BMC user encoded with base64. |
+| password | Password for the above user encoded with base64. |
+
+##### Notes
+
+The BMC address format is `<protocol>://<fqdn_or_ip_address>[:port]/[redfish-system-id]`. Some of the examples are
+
+* `redfish-virtualmedia://compute-bmc.foo.bar/redfish/v1/Systems/<uuid>`
+* `idrac://compute-bmc.subdomain.domain`
+* `idrac-virtualmedia://compute-bmc.foo.bar/redfish/v1/Systems/<uuid>`
+* `redfish://compute-bmc.foo.bar/redfish/v1/Systems/<uuid>`
+
+Additional information can be found [here](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#bmc)
+
+### Supported keys in cifmw_devscripts_extra_net_infra
+
+| Key | Description |
+| --- | ----------- |
+| name | The name of the network bridge. |
+| type | The type of network connection to be created. |
+| interface-name | The name of the interface to bind the connection. |
+| ipv4_address | IPv4 address to be applied for this connection. |
+| ipv6_address | IPv6 address to be applied for this connection. |
+
+## Examples
+
+* Sample config for deploying a compact OpenShift platform with extra nodes, existing external network,
+  separate NIC for RH-OSP networks and with OpenShift provisioning network disabled.
+
+  ```yaml
+  ---
+  ...
+  cifmw_use_devscripts: True
+
+  cifmw_devscripts_ci_token: REDACTED
+  cifmw_devscripts_pull_secret: |
+    REDACTED
+  cifmw_devscripts_dns_entries:
+    api: REDACTED
+    ingress: REDACTED
+
+  cifmw_devscripts_src_dir: "/home/ciuser/src/dev-scripts"
+
+  cifmw_devscripts_ocp_version: '4.13.13'
+  cifmw_devscripts_crb_repo: 'https://mirror.stream.centos.org/9-stream/CRB/x86_64/os/'
+  ...
+  ```
+
+* Sample config for deploying a HA OpenShift platform with additional networks, OpenShift provisioning network
+  is enabled and separate RH OSP networks.
+
+  ```YAML
+  cifmw_use_devscripts: True
+
+  cifmw_devscripts_ci_token: REDACTED
+  cifmw_devscripts_pull_secret: |
+    REDACTED
+  cifmw_devscripts_dns_entries:
+    api: REDACTED
+    ingress: REDACTED
+
+  cifmw_devscripts_ocp_version: '4.13.13'
+
+  cifmw_devscripts_config_overrides:
+    provisioning_network_profile: "Managed"
+    provisioning_network: "172.22.0.0/16"
+    num_workers: 3
+    worker_memory: 16384
+    worker_disk: 100
+    worker_vcpu: 10
+
+  cifmw_devscripts_extra_net_infra:
+    - name: "osp_trunk"
+      type: "bridge"
+      address: "192.168.122.1/24"
```

Comment:
The documentation says `ipv4_address` or `ipv6_address`. Is it a typo here?

Reply:
Yes, `address` is not a supported key. Thank you for spotting it... updating it and the below lines accordingly in the next patch set.